src/toast.toasts.ts

```diff
@@ -0,0 +1,350 @@
+                variant="tertiary"
+                @click=${this.#onToastDismissButtonClick.bind(this, toast)}
+              >
+                ${xIcon}
+              </glide-core-icon-button>
+
+              ${when(toast.privateDescription, () => {
+                return html`
+                  <div
+                    class="description"
+                    data-test="description"
+                    id="description"
+                    @click=${this.#onDescriptionClick.bind(this, toast)}
+                  >
+                    ${unsafeHTML(toast.privateDescription)}
+                  </div>
+                `;
+              })}
+            </div>`;
+          },
+        )}
+      </div>
+    </div>`;
+  }
+
+  override showPopover() {
+    this.#componentElementRef.value?.showPopover();
+  }
+
+  static async privateAdd(toast: GlideCoreToast) {
+    let toasts = document.querySelector('glide-core-private-toasts');
+
+    if (!toasts) {
+      toasts = document.createElement('glide-core-private-toasts');
+      document.body.append(toasts);
+    }
+
+    if (toast.childNodes.length > 0) {
+      let description = '';
+
+      // Toast's inner HTML will include comments (`<!--?lit$277222571$-->`) used by
+      // Lit in rendering. There's a good chance Lit will misbehave if we copy those
+      // comments. So, to be safe, we only copy Element and Text nodes.
+      for (const node of toast.childNodes) {
+        if (node instanceof Element) {
+          description += node.outerHTML;
+        } else if (node instanceof Text) {
+          description += node.textContent;
+        }
+      }
+
+      toast.privateDescription = description.trim();
+    }
+
+    toasts.toasts = [toast, ...toasts.toasts];
+
+    // The transition won't play if the toast is added to the DOM in the same
+    // tick that the `.open` class is added to `.toast`. So we first add the
+    // toast to the DOM (above), then we set `privateShow` to apply the class.
+    // Then we force a render in the next frame (below).
+    await toasts.updateComplete;
+
+    toasts.showPopover();
+    toast.privateShow = true;
+
+    setTimeout(() => {
+      // Because `privateShow` is only a reactive property of Toast, Toasts won't
+      // know it needs to rerender. So we let it know.
+      toasts.requestUpdate();
+    });
+
+    if (toast.duration < Number.POSITIVE_INFINITY) {
+      toast.privateTimeoutId = setTimeout(() => {
+        const isReducedMotion = window.matchMedia(
+          '(prefers-reduced-motion: reduce)',
+        ).matches;
+
+        if (isReducedMotion) {
+          GlideCoreToasts.privateRemove(toast);
+        } else {
+          toast.privateHide = true;
+
+          // Because `privateHide` is only a reactive property of Toast, Toasts won't
+          // know it needs to rerender. So we let it know.
+          toasts.requestUpdate();
+        }
+      }, toast.duration);
+    }
+  }
+
+  static privateRemove(toast: GlideCoreToast) {
```

Comment:
Similar to `privateAdd()` except, unlike `privateAdd()` its used by Toasts because control of the Toast has been handed over to Toasts.